### PR TITLE
PWN-3100 - Selection mechanism to pay fee in non-sol/sol token

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket (Mandatory)

https://p2pvalidator.atlassian.net/browse/PWN-3100

## Description of Work (Mandatory)

Check that the fee payer token has enough funds to pay for the fee. try to check that the fee can be paid with the user SOL token, then check it with a non-sol source token.

## Screenshots (Optional)

Screenshot #1 | Screenshot #2 | Screenshot #3
:------------:|:-------------:|:------------:
[tbd | [tbd] | [tbd]

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/send/interactor/SendInteractor.kt [hunk 1]
```diff
@@ -65,24 +70,24 @@ class SendInteractor(
...
         networkType: NetworkType
     ): FeeRelayerSendFee? {
-
         return when (networkType) {
-            NetworkType.BITCOIN ->
+            NetworkType.BITCOIN -> {
                 FeeRelayerSendFee(
-                    feeInSol = BigInteger.valueOf(20000L),
-                    feeInPayingToken = null
+                    solFee = AmountInLamports(BigInteger.valueOf(20000L)),
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/send/interactor/SendInteractor.kt:80]: to const do not know name =)
BITCOIN_FEE = 20000L
solFee = AmountInLamports(BITCOIN_FEE.toBigInteger()),

File: app/src/main/java/org/p2p/wallet/swap/interactor/orca/OrcaSwapInteractor.kt [hunk 2]
```diff
@@ -177,10 +185,47 @@ class OrcaSwapInteractor(
...
+            FeePayerUpdateResult(isUpdated = false)
+        } else {
+            setFeePayerToken(newFeePayer)
+            FeePayerUpdateResult(isUpdated = true)
+        }
+    }
+
+    private suspend fun canUserPayFeeWithSol(totalSwapFee: AmountInLamports): Boolean {
+        // если у пользователя не хватает этих же токенов для полной оплаты fee
+        // пробуем оплатить через SOL
```
Comment (kuznetsov) [inline app/src/main/java/org/p2p/wallet/swap/interactor/orca/OrcaSwapInteractor.kt:222]: Translate comments to english please

**Review by kuznetsov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/send/ui/main/SendPresenter.kt [hunk 3]
```diff
@@ -615,11 +636,20 @@ class SendPresenter(
         setTargetResult(first)
     }
 
-    private fun updateButtonText(source: Token.Active) {
+    private fun updateButtonText(source: Token.Active, sendFee: FeeRelayerSendFee?) {
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/send/ui/main/SendPresenter.kt:639]: sendFee: FeeRelayerSendFee? = null and do not make calls like
 updateButtonText(token, sendFee = null)

File: app/src/main/java/org/p2p/wallet/swap/interactor/orca/OrcaSwapInteractor.kt [hunk 4]
```diff
@@ -249,19 +301,17 @@ class OrcaSwapInteractor(
...
-    ): BigInteger {
+    suspend fun getFeesInPayingToken(feeInSOL: AmountInLamports): AmountInLamports {
         if (feePayerToken.isSOL) return feeInSOL
 
         return feeRelayerInteractor.calculateFeeInPayingToken(
-            feeInSOL = FeeAmount(accountBalances = feeInSOL),
+            feeInSOL = FeeAmount(accountBalances = feeInSOL.amount),
             payingFeeTokenMint = feePayerToken.mintAddress
-        ).total
+        ).total.let { AmountInLamports(it) }
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/swap/interactor/orca/OrcaSwapInteractor.kt:310]: let -> map ?

File: app/src/main/java/org/p2p/wallet/send/ui/main/SendPresenter.kt (hunk 3 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/send/ui/main/SendPresenter.kt:639]: im okay with explicit sense of telling that I pass fee as null

**Comment by 1sakov** [comment]
Working on this issue here - https://github.com/p2p-org/p2p-wallet-android/pull/507

**Pull request closed without merging**